### PR TITLE
Add System.Diagnostics.StackTrace tests

### PR DESCRIPTION
**Pull request by hughbe**
Noticed this project was missing a test project, so went ahead and added a bunch of tests.

### CODE REVIEW

File: src/System.Diagnostics.StackTrace/tests/Configurations.props [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <BuildConfigurations>
+      uapaot-Windows_NT;
```
Comment (joperezr) [inline src/System.Diagnostics.StackTrace/tests/Configurations.props]: no need to add all of these. It looks by checking your .csproj that the only special case is netfx, so I would have it like:
```xml
<BuildConfigurations>
  netstandard;
  netfx;
</BuildConfigurations>
```
By doing that, then the netstandard configuration will be built everywhere except on netfx, which has a specific configuration. If you are interested on how our configurations work and how do we pick each one let me know and I can gladly help :smile:

File: src/System.Diagnostics.StackTrace/tests/StackFrameTests.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        public void OffsetUnknown_Get_ReturnsNegativeOne()
+        {
+            Assert.Equal(-1, StackFrame.OFFSET_UNKNOWN);
+        }
+
+        [Fact]
+        public void Ctor_Default()
+        {
+            var stackFrame = new StackFrame();
+            VerifyStackFrame(stackFrame, false, 0, typeof(StackFrameTests).GetMethod(nameof(Ctor_Default)));
```
Comment (joperezr) [inline src/System.Diagnostics.StackTrace/tests/StackFrameTests.cs]: FYI: Reflection behaves differently in uapaot, so I'll run your tests in that platform just to make sure that they all pass.

**Comment by joperezr** [comment]
@hughbe are you able to repro the test failures that we are seeing in Jenkins?

**Comment by hughbe** [comment]
Yes and no - I haven't got a Unix VM so have basically been using the CI, sorry.

I can test Windows and netfx locally, but things are broken on Windows so I might need to invest some time setting up a VM

File: src/System.Diagnostics.StackTrace/tests/Configurations.props (hunk 1 above)
Comment (hughbe) [inline src/System.Diagnostics.StackTrace/tests/Configurations.props]: > By doing that, then the netstandard configuration will be built everywhere except on netfx, which has a specific configuration.

Oh! I understand now. That makes sense. My understanding was that you needed to specify each one, but netcoreapp somehow had some magic to connect it to netstandard if there is no netcoreapp configuration

**Comment by hughbe** [comment]
Test Innerloop netfx Windows_NT Release x64 Build and Test

**Comment by hughbe** [comment]
Amazing news - seems like everything passes the tests fine... but do we need to file an issue to consolidate the behaviour between netcoreapp and netfx?

**Comment by hughbe** [comment]
Test NETFX x86 Release Build
Test Windows x86 Release Build

**Comment by hughbe** [comment]
Test NETFX x86 Release Build

**Comment by hughbe** [comment]
Test NETFX x86 Release Build

**Comment by hughbe** [comment]
Test NETFX x86 Release Build

**Comment by danmosemsft** [comment]
@joperezr is this good to merge

File: src/System.Diagnostics.StackTrace/System.Diagnostics.StackTrace.sln [hunk 3]
```diff
@@ -15,26 +15,105 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{E107E9C1-E89
...
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ref", "ref", "{2E666815-2EDB-464B-9DF6-380BF4789AD4}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{976C7CF7-C0E6-4FD2-99FD-3A9FD3B7077A}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Diagnostics.StackTrace.Tests", "tests\System.Diagnostics.StackTrace.Tests.csproj", "{297A9116-1005-499D-A895-2063D03E4C94}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
+		netcoreapp-Unix-Debug|Any CPU = netcoreapp-Unix-Debug|Any CPU
```
Comment (joperezr) [inline src/System.Diagnostics.StackTrace/System.Diagnostics.StackTrace.sln]: Should we update this configurations, now that we move to only have two?

File: src/System.Diagnostics.StackTrace/tests/System.Diagnostics.StackTrace.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <ProjectGuid>{297A9116-1005-499D-A895-2063D03E4C94}</ProjectGuid>
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the configurations -->
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uapaot-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uapaot-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Windows_NT-Release|AnyCPU'" />
```
Comment (joperezr) [inline src/System.Diagnostics.StackTrace/tests/System.Diagnostics.StackTrace.Tests.csproj]: Same here, lets remove unused configurations

**Comment by joperezr** [comment]
StackTrace is not supported in uapaot so that means that all `System.Diagnostics.Tests.StackFrameExtensionsTests`, `System.Diagnostics.Tests.StackTraceTests` and `System.Diagnostics.Tests.StackFrameTests` are all failing with NotImplementedException. Do you mind adding a `SkipOnTargetFramework` attribute for Uapaot for those test classes?

**Comment by hughbe** [comment]
I've updated the PR to skip the failing uapaot tests

**Comment by danmosemsft** [comment]
LGTM thanks @hughbe !

**Pull request merged**